### PR TITLE
Security basics

### PR DESCRIPTION
**Pull request by mykolaskrypka**

### CODE REVIEW

File: src/main/java/mate/academy/controller/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+@RequestMapping("api/auth")
+public class AuthenticationController {
+    private final UserService userService;
+
+    @Operation(summary = "Registration", description = "User registration")
+    @PostMapping("/registration")
+    public ResponseEntity<UserResponseDto> register(
+            @RequestBody @Valid UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        return new ResponseEntity<>(userService.register(requestDto), HttpStatus.OK);
```
Comment (sokimaaa) [inline src/main/java/mate/academy/controller/AuthenticationController.java]:
```suggestion
        return ResponseEntity.ok(body);
```

File: src/main/java/mate/academy/exception/RegistrationException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package mate.academy.exception;
+
+public class RegistrationException extends Exception {
```
Comment (sokimaaa) [inline src/main/java/mate/academy/exception/RegistrationException.java]: prefer using unchecked exception over checked.

File: src/main/java/mate/academy/repository/SpecificationBuilder.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package mate.academy.repository;
+
+import mate.academy.dto.BookSearchParameters;
+import org.springframework.data.jpa.domain.Specification;
+
+public interface SpecificationBuilder<T> {
+    Specification<T> build(BookSearchParameters searchParameters);
```
Comment (sokimaaa) [inline src/main/java/mate/academy/repository/SpecificationBuilder.java:7]: This class is not a part of current task.

File: src/main/java/mate/academy/repository/SpecificationProvider.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package mate.academy.repository;
+
+import org.springframework.data.jpa.domain.Specification;
+
+public interface SpecificationProvider<T> {
+    String getKey();
```
Comment (sokimaaa) [inline src/main/java/mate/academy/repository/SpecificationProvider.java:6]: This class is not a part of current task.

File: src/main/java/mate/academy/repository/user/UserRepository.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package mate.academy.repository.user;
+
+import java.util.Optional;
+import mate.academy.model.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+    @Override
+    User save(User user);
```
Comment (sokimaaa) [inline src/main/java/mate/academy/repository/user/UserRepository.java]:
```suggestion
```

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-books-table.yaml
+  - include:
+      file: db/changelog/changes/02-adding-records-into-books-table.yaml
+  - include:
+      file: db/changelog/changes/03-create-users-table.yaml
```
Comment (sokimaaa) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: consider inserting default users

File: src/main/java/mate/academy/repository/SpecificationBuilder.java (hunk 3 above)
Comment (mykolaskrypka) [reply src/main/java/mate/academy/repository/SpecificationBuilder.java:7]: 100% it is from Criteria Query in SpringBoot

File: src/main/java/mate/academy/repository/SpecificationProvider.java (hunk 4 above)
Comment (mykolaskrypka) [reply src/main/java/mate/academy/repository/SpecificationProvider.java:6]: 100% it is from Criteria Query in SpringBoot

**Pull request closed without merging**